website_snippet_data_slider/static/src/js/data_slider.js

```diff
@@ -0,0 +1,134 @@
+/* Copyright 2016 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+ */
+
+odoo.define('website_snippet_data_slider', function(require){
+  "use strict";
+    
+    var animation = require('website.content.snippets.animation');
+    
+    var defaults = {
+        lazyLoad: 'ondemand',
+        slidesToShow: 5,
+        slidesToScroll: 1,
+        dots: true,
+        infinite: true,
+        speed: 500,
+        arrows: true,
+        autoplay: true,
+        adaptiveHeight: false,
+        variableWidth: false,
+        autoplaySpeed: 3000,
+        data_model: 'product.template',
+        data_domain: [['website_published', '=', true]],
+        data_image_field: 'image_medium',
+        data_name_field: 'display_name',
+        data_title: 'Featured Products',
+        data_title_tag: 'h1',
+        data_title_class: 'text-center',
+        data_uri_field: 'website_url',
+        data_container_width: '90%',
+        data_limit: 10,
+        prevArrow: '<a href="#" class="slider-arrow-left"><i class="fa fa-arrow-left fa-2x"></i></a>',
+        nextArrow: '<a href="#" class="slider-arrow-right"><i class="fa fa-arrow-right fa-2x"></i></a>',
+    };
+    
+    animation.registry.data_slider = animation.Class.extend({
+        selector: ".o_data_slider",
+        
+        slickSetOption: function(event, key, val) {
+            switch (val) {
+                case 'true':
+                    val = true;
+                    break;
+                case 'false':
+                    val = false;
+                    break;
+                case undefined:
+                    return;
+            }
+            if (typeof val === 'object') {
+                return;
+            }
+            this.$slick.slick('slickSetOption', key, val, true);
+        },
+        
+        // It loops parses JSON records and calls _handleRecord on each
+        handleRecords: function(records) {
+            _.each(JSON.parse(records), $.proxy(this._handleRecord, this)); 
+        },
+        
+        // Accepts a record object and appends to slick
+        _handleRecord: function(record) {
+            var $img = $('<img>');
```

Comment:
I'd add a TODO here to replace this element creation w/ pure JS: much more performant.